### PR TITLE
<Image /> Component

### PR DESCRIPTION
**Pull request by Zombiefruit**

### CODE REVIEW

File: src/components/image/image.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    @observable src: string = this.props.src ? this.props.src : this.props.defaultImage!;
+
+    @action setSrcToFallback = () => {
+        this.src = this.getFallbackSrcFor(this.src)!;
+    };
+
+    onError: React.EventHandler<SyntheticEvent<HTMLImageElement>> = () => {
+        if (this.props.onError) {
+            this.props.onError({data: this.src});
```
Comment (wtfil) [inline src/components/image/image.tsx]: The best practice is to use instance of error as argument of `onError`'s like functions.

You can create own error class if you need to provide more than stack and message
```ts
class ImageError extends Error {
    constructor(src) { 
        super('Image Error')
        this.src = src
    }
    src: string
 }
```
and inside `onError` you can call `onError(new ImageError(this.src))`

Comment (Zombiefruit) [inline src/components/image/image.tsx]: Cool, will do

Comment (Zombiefruit) [inline src/components/image/image.tsx]: Cool, thanks for the advice :D. I'll make the change

File: test/components/image.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { expect, ClientRenderer, sinon, simulate, waitForDom, waitFor } from 'test-drive-react';
+import { Image } from '../../src';
+import { objectFitSupported } from '../../src/common/environment';
+import { onePixelTransparentSrc } from '../../src';
+import { sampleImage } from '../fixtures/sample-image';
+
+const nativeImage = 'NATIVE_IMAGE';
+
+describe('<Image />', () => {
```
Comment (AviVahl) [inline test/components/image.spec.tsx]: async/await all async tests.

File: src/components/image/image.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        // first, fallback to defaultImage, and later to one transparent pixel
+        return (src !== this.props.defaultImage) ? this.props.defaultImage : onePixelTransparentSrc;
+    }
+
+    render() {
+        // remove certain props from the received props that shouldn't be applied to image tag
+        const { title, alt, defaultImage, onLoadError, ...rest } = this.props;
+
+        return (
+            <img {...rest}
```
Comment (tyv) [inline src/components/image/image.tsx]: I suppose `{...rest}` here for reason: because some props are not allowed for native evelements

Comment (AviVahl) [inline src/components/image/image.tsx]: @Zombiefruit and I got a nice new pattern going for the props after working on this a bit together.
We overloaded onError and onLoad while passing on the original events (with src attached) and removing the need for the different onLoadError. He'll commit and we'll keep working on this today, as the current implementation is missing functionality on browsers on supporting object-fit.

File: src/common/null-function.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export function nullFunction() {}
```
Comment (tyv) [inline src/common/null-function.ts]: usually it says `noop`

File: src/components/image/image.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import { nullFunction } from '../../common/null-function';
+
+// Transparent 1x1 gif image
+export const onePixelTransparentSrc = 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';
```
Comment (tyv) [inline src/components/image/image.tsx]: maybe good place for that is `./common`

**Pull request merged**